docs/deployment/universal-profile.md

```diff
@@ -159,31 +200,242 @@ await lspFactory.LSP3UniversalProfile.deploy({
+
+`deployProxy` defaults to `true` for `ERC725Account` and `LSP6KeyManager` and `false` for `UniversalReceiverDelegate`.
+
+When using proxy deployment, LSPFactory will check that there is bytecode deployed at the base contract address before deploying. If there is none, a new base contract will be deployed and referenced in the proxy contract. This is useful when using LSPFactory on a local development network like Hardhat where there will be no pre-deployed base contracts.
+
+#### Universal Receiver Delegate proxy deployment
+
+The `UniversalReceiverDelegate` is a logic contract which writes to the Universal Profile when it recieves some asset. This is not specific to any particular Universal Profile so the same `UniversalReceiverDelegate` contract can be used for multiple different Universal Profile deployments.
+
+By default LSPFactory will use the latest available version of `UniversalReceiverDelegate` version stored in the [version file](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json). This address is used directly on the Universal Profile and is given the `SETDATA` LSP6 permission. This means that no `UniversalReceiverDelegate` contract needs to be deployed when deploying a Universal Profile which further saves on the gas cost of Universal Profile deployment.
+
+To specify that your Universal Profile should use proxy deployment when deplying the `UniversalReceiverDelegate` contract, set `deployProxy` to `true`.
+
+```javascript
+lspFactory.LSP3UniversalProfile.deploy({...}, {
+    UniversalReceiverDelegate: {
+        deployProxy: true,
+    },
+})
+```
+
+### Using a Custom Address
+
+You can specify the base contract address of a contract by passing the `libAddress` parameter. This allows you to use a custom implementation of a contract standard by using a custom base contract you have previously deployed. A custom base contract passed here must adhere to the relevant LSP contract standard it is being used for.
+
+```javascript title="Deploying a Universal Profile with a custom base contract implementation of ERC725Account"
+lspFactory.LSP3UniversalProfile.deploy({...}, {
+    ERC725Account: {
+        libAddress: '0x00b1d454Eb5d917253FD6cb4D5560dEC30b0960c',
+    },
+})
+```
+
+:::info Info
+`UniversalReceiverDelegate` does not use proxy deployment by default. When setting the `UniversalReceiverDelegate` `libAddress`, if `deployProxy` is not set to `true` LSPFactory will use the provided address directly. The provided address will be given the `SETDATA` LSP6 permission and set as the [LSP1UniversalReceiverDelegate key](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md#lsp1universalreceiverdelegate) on the ERC725Account. [Read more](./universal-profile#universal-receiver-delegate-proxy-deployment)
```

Comment:
Ah ok I see. You can specify proxy deployment for the URD. But by default, it is set to `false`, correct? :)

Reply:
yes exactly :) Some people may have a particular use case, or just want their URD to be dedicated to their UP